File: test_projects/qwen_qwen_2_5_coder_32b_instruct_java/UserController.java

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;

import java.util.List;

@RestController
@RequestMapping("/users")
public class UserController {
    @Autowired
    private UserRepository userRepository;

    @GetMapping
    public ResponseEntity<List<User>> getAllUsers() {
        return new ResponseEntity<>(userRepository.findAll(), HttpStatus.OK);
    }

    @GetMapping("/{id}")
    public ResponseEntity<User> getUserById(@PathVariable int id) {
        User user = userRepository.findById(id);
        if (user != null) {
            return new ResponseEntity<>(user, HttpStatus.OK);
        } else {
            return new ResponseEntity<>(HttpStatus.NOT_FOUND);
        }
    }

    @PostMapping
    public ResponseEntity<User> createUser(@RequestBody User user) {
        userRepository.save(user);
        // Assuming last_insert_id() function is available in PostgreSQL to get the last inserted id
        int newId = userRepository.jdbcTemplate.queryForObject("SELECT lastval()", Integer.class);
        user.setId(newId);
        return new ResponseEntity<>(user, HttpStatus.CREATED);
    }

    @PutMapping("/{id}")
    public ResponseEntity<Void> updateUser(@PathVariable int id, @RequestBody User user) {
        int updatedRows = userRepository.update(id, user);
        if (updatedRows == 1) {
            return new ResponseEntity<>(HttpStatus.NO_CONTENT);
        } else {
            return new ResponseEntity<>(HttpStatus.NOT_FOUND);
        }
    }

    @DeleteMapping("/{id}")
    public ResponseEntity<Void> deleteUser(@PathVariable int id) {
        int deletedRows = userRepository.delete(id);
        if (deletedRows == 1) {
            return new ResponseEntity<>(HttpStatus.NO_CONTENT);
        } else {
            return new ResponseEntity<>(HttpStatus.NOT_FOUND);
        }
    }
}

File: test_projects/google_gemini_2_0_flash_001_kotlin/bin/main/com/example/UserController.kt

package com.example

import org.springframework.http.HttpStatus
import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.*
import org.springframework.jdbc.core.JdbcTemplate
import org.springframework.web.server.ResponseStatusException
import org.springframework.dao.EmptyResultDataAccessException
import org.springframework.http.MediaType
import java.net.URI

@RestController
@RequestMapping("/users", produces = [MediaType.APPLICATION_JSON_VALUE])
class UserController(private val jdbcTemplate: JdbcTemplate) {

    @PostMapping
    fun createUser(@RequestBody user: User): ResponseEntity<User> {
        val sql = "INSERT INTO users (name, email) VALUES (?, ?) RETURNING id"
        val id = jdbcTemplate.queryForObject(sql, arrayOf(user.name, user.email), Int::class.java)
            ?: throw IllegalStateException("Failed to insert user")

        val createdUser = User(id, user.name, user.email)
        return ResponseEntity.created(URI.create("/users/$id")).body(createdUser)
    }

    @GetMapping
    fun getAllUsers(): List<User> {
        val sql = "SELECT id, name, email FROM users"
        return jdbcTemplate.query(sql) { rs, _ ->
            User(rs.getInt("id"), rs.getString("name"), rs.getString("email"))
        }
    }

    @GetMapping("/{id}")
    fun getUserById(@PathVariable id: Int): User {
        val sql = "SELECT id, name, email FROM users WHERE id = ?"
        return try {
            jdbcTemplate.queryForObject(sql, arrayOf(id)) { rs, _ ->
                User(rs.getInt("id"), rs.getString("name"), rs.getString("email"))
            } ?: throw ResponseStatusException(HttpStatus.NOT_FOUND, "User not found")
        } catch (e: EmptyResultDataAccessException) {
            throw ResponseStatusException(HttpStatus.NOT_FOUND, "User not found")
        }
    }

    @PutMapping("/{id}")
    fun updateUser(@PathVariable id: Int, @RequestBody user: User): ResponseEntity<Any> {
        val sql = "UPDATE users SET name = ?, email = ? WHERE id = ?"
        val rowsAffected = jdbcTemplate.update(sql, user.name, user.email, id)
        return if (rowsAffected > 0) {
            ResponseEntity.noContent().build()
        } else {
            throw ResponseStatusException(HttpStatus.NOT_FOUND, "User not found")
        }
    }

    @DeleteMapping("/{id}")
    fun deleteUser(@PathVariable id: Int): ResponseEntity<Any> {
        val sql = "DELETE FROM users WHERE id = ?"
        val rowsAffected = jdbcTemplate.update(sql, id)
        return if (rowsAffected > 0) {
            ResponseEntity.noContent().build()
        } else {
            throw ResponseStatusException(HttpStatus.NOT_FOUND, "User not found")
        }
    }
}
